packer/builder/azure/smapi/config_test.go

```diff
@@ -69,24 +69,48 @@ func TestConfig_VMImageSource(t *testing.T) {
 	defer os.Remove(f)
 	log.SetOutput(testLogger{t}) // hide log if test is succes
 
+	m := make(map[string]string)
+	m["name"] = "Ubuntu-14_04_3-LTS-amd64-server-20160119-en-us-30GB"
+	m["link"] = "http://www.microsoft.com/"
+
 	tcs := []struct {
 		cfgmod func(map[string]interface{})
+		msg    string
 		err    bool
 	}{
-		{func(cfg map[string]interface{}) { cfg["remote_source_image_link"] = "http://www.microsoft.com/" }, true},
-		{func(cfg map[string]interface{}) {
-			cfg["remote_source_image_link"] = "http://www.microsoft.com/"
+		{func(cfg map[string]interface{}) { // none
 			delete(cfg, "os_image_label")
-		}, false},
-		{func(cfg map[string]interface{}) { delete(cfg, "os_image_label") }, true},
+		}, "No defined source", true},
+		{func(cfg map[string]interface{}) { // label and link
+			cfg["remote_source_image_link"] = m["link"]
+		}, "Both label and link defined", true},
+		{func(cfg map[string]interface{}) { // label and name
+			cfg["os_image_name"] = m["name"]
+		}, "Both label and name defined", true},
+		{func(cfg map[string]interface{}) { // label and name and link
+			cfg["os_image_name"] = m["name"]
+			cfg["remote_source_image_link"] = m["link"]
+		}, "Both name and link defined", true},
```

Comment:
It's fine. I was just pointing out that 'label' is also defined in this test case (because it's defined by the setup method and not deleted in this func). But it's fine... :-)